### ISSUE TITLE
Diego Cells fail on post-start script due to Garden timeouts - "Timed out pinging garden server"

### ISSUE THREAD

**Issue by Hermen-Nicolau** [cloudfoundry/garden-runc-release#153]
# Description
In the middle of rotating nonconfigurable leaf certs. The customer that I'm working with used  "om regenerate-certificates" and started applying opsman changes. The ensuing bosh deployment of PAS went fine up until getting to Diego Cells. Most diego cells are updated fine but several end up failing the post-start script. Running the deployment again immediately after, the same error occurs. As a workaround, this issue can be resolved by SSHing onto the problematic diego cell and running "sudo reboot"- and then deploy again- at which point bosh updates that diego cell fine. However, bosh will then get stuck on the same error with another diego cell.

From the logs provided we can see that garden "starts". However, we don’t ever seem to see this line:
https://github.com/cloudfoundry/guardian/blob/aa361ad5867b6105dc5c9771a23e066609c57cbe/gardener/gardener.go#L579
gardener/gardener.go:579
`destroyLog.Info("cleaned-up"`)
The hypothesis is that something is stuck cleaning. 

## Environment

* garden-runc-release version: garden-runc-1.19.5-ubuntu-xenial-315.93
* Enabled configuration: 
* IaaS: vSphere 
* Stemcell version: Garden runc --> ubuntu-xenial-315.93 | PAS --> ubuntu-xenial-315.99
* Kernel version: 4.15.0-64-generic

## Steps to reproduce
So far this issue has happening in every 2.6.5 PCF foundation that the customer rotates nonconfigurable leaf certs. It has happened in 3 different foundations already. Recently, they upgraded 1 of those same environments to 2.7.7., abd had zero issues with the Diego Cells and/or Garden. 

## Logs
From the Change logs on opsman after the apply changes:
`Error: Action Failed get_task: Task 4df1dcad-a399-4627-6149-1921ddd2ce3b result: 2 of 6 post-start scripts failed. Failed Jobs: garden, rep.`

Garden post-start.stdout.log:
```
Tue Jan 14 15:13:16 UTC 2020: Pinging garden server...
Tue Jan 14 15:13:16 UTC 2020: Attempt 1..
.
.
.
Tue Jan 14 15:15:16 UTC 2020: Timed out pinging garden server.
```
Rep rep.stdout.log
`{"timestamp":"1579014702.231879473","source":"rep","message":"rep.wait-for-garden.failed-to-ping-garden","log_level":2,"data":{"error":"Get http://api/ping: dial unix /var/vcap/data/garden/garden.sock: connect: no such file or directory","initialTime:":"2020-01-14T15:11:42.231338759Z","session":"1"}}`

garden starts:
`"timestamp":"1579014695.068500996","source":"guardian","message":"guardian.start.starting","log_level":1,"data":{"session":"4"}}
`

Dontpanic logs are too big to be attached. 

## Cause
We don't know the cause for this issue but it has happened in more that one foundation 

## Resolution
Per the timestamps, during the error the file /var/vcap/data/garden/garden.sock does not exist and that seems to be the reason why we see the timeouts and consequently the failures in the post-start script.  As a workaround, we have to ssh into the affected Diego Cell and run a "sudo reboot" which seems to create the file /var/vcap/data/garden/garden.sock.

**Comment by danail-branekov** [comment]
Hey @Hermen-Nicolau 

According to the dontpanic report I am pretty sure that you are facing bug #151 . We fixed that with GRR 1.19.8, unfortunately you are still on 1.19.5.

In order to make very-very sure that this is a duplicate indeed, could you please ssh onto an affected cell and perform the following checks:
* Run `ps auxf | grep groot` - there should be at least two grootfs processes trying to delete one and the same image (the image ID is the last `grootfs delete` argument)
* Verify that the grootfs processes are in `S` state
* Try listing the image rootfs directory via `ls -la /var/vcap/data/grootfs/store/unprivileged/images/<image_id>/rootfs`. Do you see the `.` and the `..` entries in the output? If not, this issue is definitely a duplicate of #151 


Thanks!

**Comment by danail-branekov** [comment]
As discussed over Slack, the VMs are rebooted and the problematic mount is gone. I am closing the issue, please reopen if it turns out to be something else.
